.github/workflows/pull_requests.yml

```diff
@@ -0,0 +1,31 @@
+name: Pull Request
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, ready_for_review]
+
+## We are cancelling previously triggered workflow runs
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.ref }}
```

Comment:
`${{ github.event.ref }}` is not quite right. It should be `${{ github.ref }}`.
`github.event` gives you access to the full webhook event which can be anything and doesn't necessarily contain a `ref`. As a matter of fact it doesn't contain a `ref` for pull-request events.

`${{ github.ref }}` on the other hand is another GitHub-injected value, just like `${{ github.event }}`, and is guaranteed to exist.